### PR TITLE
Automatically add configuration file after bot installation [DRAFT]

### PR DESCRIPTION
**Pull request by twisstosin**
title: Automatically add configuration file after bot installation
description: This will enable collectives to get started with bot configuration quickly by automatically creating a pull request containing the bot configuration file and basic setting with can be edited by user.

### CODE REVIEW

**Comment by znarf** [comment]
`yarn.lock` is updated while `package.json` is untouched, can you check that?

**Comment by twisstosin** [comment]
Yeah, It was changed and I'll add it in the next commit, Thanks.

File: src/bot.ts [hunk 1]
```diff
@@ -18,6 +18,82 @@ import { is, not } from './utils'
...
+    //get 'added' action as condition before adding the config file
+    if (context.payload.action === 'added') {
+      let added_repos = context.payload.repositories_added
+
+      // This can open PRs with new files at any point in time, but this
+      // gets called when the app is first installed on a repo
+      context.log({ event: context.event, action: context.payload.action })
+
+      let fields = {
+        file: {
```
Comment (znarf) [inline src/bot.ts:49]: Maybe we should extract the yaml content in a separated file that we just read from the filesystem here.

Some ideas:
- most if not all options should contains the default values, maybe commented by default
- the file should be heavily commented to explain the different options to the user (I'll try to provide a good example of that, I have PHP or elasticsearch configuration?).

Comment (twisstosin) [reply src/bot.ts:49]: Yeah, extracting to a separate file would be much better, I'll do that.

I'll try to add the default values, and an example would be very helpful, thanks.

Comment (znarf) [reply src/bot.ts:49]: I was mentioning such configuration files. I don't think they are the best, but they're making a good job at explaining the features, and in the case of elasticsearch.yml the defaults.

- https://github.com/elastic/elasticsearch/blob/master/distribution/src/config/elasticsearch.yml
- https://github.com/php/php-src/blob/master/php.ini-production

**Comment by znarf** [comment]
@twisstosin is it testable now?

File: src/bot.ts [hunk 2]
```diff
@@ -18,6 +18,82 @@ import { is, not } from './utils'
...
+      const branch = `add-${fields.file.path}` // your branch's name
+      const content = Buffer.from(fields.file.content).toString('base64') // content for your configuration file
+
+      const reference = await context.github.git.getRef({
+        owner: context.repo.owner,
+        repo: context.repo.repo,
+        ref: 'heads/master',
+      }) // get the reference for the master branch
+
+      console.log(
```
Comment (znarf) [inline src/bot.ts]: Don't use that pattern with console.log, make it clear what you are logging. When possible, single line message.

Comment (twisstosin) [reply src/bot.ts]: Alright, will update.

**Comment by twisstosin** [comment]
> @twisstosin is it testable now?

I'll push another commit, that'll be testable and with log pattern changes.

**Comment by znarf** [comment]
@twisstosin if it's working, what do you think is needed to polish and finish that feature? what do you need from us?

Also, please check the test (I triggered it for your branch here and it's not passing https://circleci.com/gh/opencollective/opencollective-bot/284) and make sure we keep 100% coverage.

**Comment by znarf** [comment]
@twisstosin Unfortunately, if not completed by the end of the week, we will have to move forward in another way for this feature.

**Comment by znarf** [comment]
Thank your for your effort! The core of the feature was ultimately implemented in https://github.com/opencollective/opencollective-bot/pull/167 It was actually more complicated than anticipated.

Let me know if you have any feedback on the feature.

**Pull request closed without merging**